## Import libraries to use for webscrapping

Will be using selenium to read for the motorycles main page includes scrapping the link in each page and beautifulsoup to scrape the data from each motorcyle profile page. 

In [39]:
import re
import time
import requests
import pandas as pd
from selenium import webdriver
from bs4 import BeautifulSoup 
from selenium.webdriver.common.by import By
from selenium.webdriver.chrome.options import Options
from webdriver_manager.chrome import ChromeDriverManager
from selenium.common.exceptions import NoSuchElementException

Putting the path in a driver path and opens the web in a new chrome window. No page is shown as no url is given.

In [374]:
DRIVER_PATH = "/path/to/chromedriver"

options = Options()
options.headless = True
options.add_argument("--window-size=1920,1200")

driver = webdriver.Chrome(ChromeDriverManager().install())

[WDM] - Downloading: 100%|██████████| 8.61M/8.61M [00:00<00:00, 35.4MB/s]
/var/folders/bq/1n8bgy_57l74n18m4pnvpyj00000gn/T/ipykernel_97374/3677721619.py:7: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(ChromeDriverManager().install())


## SGBIKEMART.COM.SG

Before getting the listings, I have to split the listings into used  and new bike as sgbikemart.com provide different links.

In [3]:
used_bike = 'https://sgbikemart.com.sg/listing/usedbikes/listing/'
new_bike = 'https://sgbikemart.com.sg/listing/newbikes/listing/'

Basic Functions for easy understanding what is happening in the loop, and will be needing it few times in this notebook.

In [46]:
def scroll_to_end():
    driver.execute_script("window.scrollTo(0, document.body.scrollHeight);")
    time.sleep(3)

def next_page():
    button = driver.find_element(By.LINK_TEXT,'next')
    button.click()
    time.sleep(3)

def go_to_page(clas):
    driver.get(clas)
    time.sleep(3)


- Scroll to the end of the page : scroll_to_end()

- Clicked the 'next page' button : next_page():

- Goes to the page that is given in the bracket() : go_to_page(clas):


A while loop to get the links from used bike, from page to page. The for loop is to get the link  in each tag name. I delete the first 19 rows and the last 25 rows as its not the bikes page link.

In [138]:
count = 0
draft = []
go_to_page(used_bike)
while driver.find_element(By.LINK_TEXT,'last') != None:
    links = []
    scroll_to_end()
    profile = driver.find_elements(By.TAG_NAME,"a")

    for link in profile:
        href = link.get_attribute('href')
        links.append(href)

    del links[:19]
    del links[-25:]
    draft.append(links)
    count += 1

    try:
        next_page()
    except NoSuchElementException as exc:
        print(exc)
        print("Total page:",count)
        break
       

Message: no such element: Unable to locate element: {"method":"link text","selector":"next"}
  (Session info: chrome=108.0.5359.124)
Stacktrace:
0   chromedriver                        0x0000000102dfff38 chromedriver + 4910904
1   chromedriver                        0x0000000102d7fa03 chromedriver + 4385283
2   chromedriver                        0x00000001029c4747 chromedriver + 472903
3   chromedriver                        0x0000000102a0934c chromedriver + 754508
4   chromedriver                        0x0000000102a095a1 chromedriver + 755105
5   chromedriver                        0x0000000102a4ce94 chromedriver + 1031828
6   chromedriver                        0x0000000102a2f13d chromedriver + 909629
7   chromedriver                        0x0000000102a4a28e chromedriver + 1020558
8   chromedriver                        0x0000000102a2eee3 chromedriver + 909027
9   chromedriver                        0x00000001029f930c chromedriver + 688908
10  chromedriver                        0

In [139]:
draft1 = draft
draft1

[['https://sgbikemart.com.sg/listing/usedbike/bmw-bmw-r1250rt/27190/',
  'https://sgbikemart.com.sg/listing/usedbike/bmw-bmw-r1250rt/27190/',
  'https://sgbikemart.com.sg/listing/usedbike/bmw-bmw-r1250rt/27190/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-vfr800x-crossrunner/27209/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-vfr800x-crossrunner/27209/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-vfr800x-crossrunner/27209/',
  'https://sgbikemart.com.sg/listing/usedbike/yamaha-yamaha-xsr900/27195/',
  'https://sgbikemart.com.sg/listing/usedbike/yamaha-yamaha-xsr900/27195/',
  'https://sgbikemart.com.sg/listing/usedbike/yamaha-yamaha-xsr900/27195/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-pcx150/27220/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-pcx150/27220/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-pcx150/27220/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-shadow-400/27219/'

Apparently i cant get the exact links one shot, i have a list of a list links, so i had to filter out the duplicate links.

In [156]:
for num in range(len(draft1)):
    if len(draft1[num]) == 66:
        del draft1[num][-6:]
    if len(draft1[num]) == 65:
        del draft1[num][-5:]
    if len(draft1[num]) == 64:
        del draft1[num][-4:]
    if len(draft1[num]) == 63:
        del draft1[num][-3:]
    if len(draft1[num]) == 62:
        del draft1[num][-2:]

60
62
63
64
65
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
66
65
64
63
62
9


In [165]:
for num in range(len(draft1)):
    if len(draft1[num]) == 60:
        del draft1[num][::3]
    if len(draft1[num]) == 9:
        del draft1[num][::3]

In [ ]:
for num in range(len(draft1)):
    if len(draft1[num]) == 40:
        del draft1[num][::2]
    if len(draft1[num]) == 7:
        del draft1[num][::2]

In [173]:
draft1

[['https://sgbikemart.com.sg/listing/usedbike/bmw-bmw-r1250rt/27190/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-vfr800x-crossrunner/27209/',
  'https://sgbikemart.com.sg/listing/usedbike/yamaha-yamaha-xsr900/27195/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-pcx150/27220/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-shadow-400/27219/',
  'https://sgbikemart.com.sg/listing/usedbike/yamaha-yamaha-fjr1300/27218/',
  'https://sgbikemart.com.sg/listing/usedbike/vespa-vespa-lx150-ie/22665/',
  'https://sgbikemart.com.sg/listing/usedbike/honda-honda-nt700v-deauville/21006/',
  'https://sgbikemart.com.sg/listing/usedbike/yamaha-yamaha-nmax-155/27217/',
  'https://sgbikemart.com.sg/listing/usedbike/ktm-ktm-1290-super-adventure/27216/',
  'https://sgbikemart.com.sg/listing/usedbike/ktm-ktm-390-duke/26540/',
  'https://sgbikemart.com.sg/listing/usedbike/vespa-vespa-gts-300-supertech-hpe/26557/',
  'https://sgbikemart.com.sg/listing/usedbike/mv-agust

Finally, got the total links of used bikes to scrapped. Below is getting the links for new bikes.

In [190]:
count = 0
draft = []
go_to_page(new_bike)
while driver.find_element(By.LINK_TEXT,'last') != None:
    links = []
    scroll_to_end()
    profile = driver.find_elements(By.TAG_NAME,"a")

    for link in profile:
        href = link.get_attribute('href')
        links.append(href)

    draft.append(links)
    count += 1

    try:
        next_page()
    except NoSuchElementException as exc:
        print(exc)
        print("Total page:",count)
        break
       

Message: no such element: Unable to locate element: {"method":"link text","selector":"next"}
  (Session info: chrome=108.0.5359.124)
Stacktrace:
0   chromedriver                        0x0000000102dfff38 chromedriver + 4910904
1   chromedriver                        0x0000000102d7fa03 chromedriver + 4385283
2   chromedriver                        0x00000001029c4747 chromedriver + 472903
3   chromedriver                        0x0000000102a0934c chromedriver + 754508
4   chromedriver                        0x0000000102a095a1 chromedriver + 755105
5   chromedriver                        0x0000000102a4ce94 chromedriver + 1031828
6   chromedriver                        0x0000000102a2f13d chromedriver + 909629
7   chromedriver                        0x0000000102a4a28e chromedriver + 1020558
8   chromedriver                        0x0000000102a2eee3 chromedriver + 909027
9   chromedriver                        0x00000001029f930c chromedriver + 688908
10  chromedriver                        0

In [194]:
draft2 = draft
draft2

[['https://sgbikemart.com.sg/',
  'https://sgbikemart.com.sg/listing/newbikes/listing/#',
  'https://sgbikemart.com.sg/listing/newbikes/dealer/',
  'https://sgbikemart.com.sg/listing/usedbikes/',
  'https://sgbikemart.com.sg/listing/accessories/',
  'https://sgbikemart.com.sg/auction/',
  'https://sgbikemart.com.sg/rent/motorcycle/',
  'https://sgbikemart.com.sg/listing/newbikes/',
  'https://sgbikemart.com.sg/motorcycle-insurance/',
  'https://sgbikemart.com.sg/directory/',
  'https://sgbikemart.com.sg/blog/',
  'https://sgbikemart.com.sg/chat/',
  'https://sgbikemart.com.sg/alerts/shortlist/',
  'https://sgbikemart.com.sg/listing/newbikes/listing/#',
  'https://sgbikemart.com.sg/listing/newbikes/listing/#',
  'https://sgbikemart.com.sg/listing/newbikes/listing/#',
  'https://sgbikemart.com.sg/sell/',
  'https://sgbikemart.com.sg/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad1-200/816/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad1-200/816/',
  'https:/

Doing the same method to new bikes too.

In [196]:
for num in range(len(draft2)):
    del draft2[num][:18]
    del draft2[num][-25:]

In [198]:
for num in range(len(draft2)):
    if len(draft2[num]) == 36:
        del draft2[num][-6:]
    if len(draft2[num]) == 35:
        del draft2[num][-5:]
    if len(draft2[num]) == 34:
        del draft2[num][-4:]
    if len(draft2[num]) == 33:
        del draft2[num][-3:]
    if len(draft2[num]) == 32:
        del draft2[num][-2:]

In [200]:
for num in range(len(draft2)):
    if len(draft2[num]) == 30:
        del draft2[num][::3]
    if len(draft2[num]) == 6:
        del draft2[num][::3]

In [203]:
for num in range(len(draft2)):
    if len(draft2[num]) == 20:
        del draft2[num][::2]
    if len(draft2[num]) == 4:
        del draft2[num][::2]

In [201]:
draft2

[['https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad1-200/816/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad1-200/816/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad2-400/817/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad2-400/817/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad3-300/263/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad3-300/263/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad3-400/818/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ad3-400/818/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ar3-200/468/',
  'https://sgbikemart.com.sg/listing/newbike/adiva-adiva-ar3-200/468/',
  'https://sgbikemart.com.sg/listing/newbike/aprilia-aprilia-dorsoduro-900/475/',
  'https://sgbikemart.com.sg/listing/newbike/aprilia-aprilia-dorsoduro-900/475/',
  'https://sgbikemart.com.sg/listing/newbike/aprilia-aprilia-rs125/1070/',
  'https://sgbikemart.com.sg/listing/newb

Getting the lengthe of list, theres 38 pages for used bikes and 90 pages for new bikes to be scrapped.

In [204]:
len(draft1)

38

In [205]:
len(draft2)

90

Putting them together in a list, for easier process.

In [206]:
motorcycle = draft1[0] + draft1[1] + draft1[2] + draft1[3] + draft1[4] + draft1[5] + draft1[6] + draft1[7] + draft1[8] + draft1[9] + draft1[10] + draft1[11] + draft1[12] + draft1[13] + draft1[14] + draft1[15] + draft1[16] + draft1[17] + draft1[18] + draft1[19] + draft1[20] + draft1[21] + draft1[22] + draft1[23] + draft1[24] + draft1[25] + draft1[26] + draft1[27] + draft1[28] + draft1[29] + draft1[30] + draft1[31] + draft1[32] + draft1[33] + draft1[34] + draft1[35] + draft1[36] + draft1[37] + draft2[0] + draft2[1] + draft2[2] + draft2[3] + draft2[4] + draft2[5] + draft2[6] + draft2[7] + draft2[8] + draft2[9] + draft2[10] + draft2[11] + draft2[12] + draft2[13] + draft2[14] + draft2[15] + draft2[16] + draft2[17] + draft2[18] + draft2[19] + draft2[20] + draft2[21] +  draft2[22] + draft2[23] + draft2[24] + draft2[25] + draft2[26] + draft2[27] + draft2[28] + draft2[29] + draft2[30] + draft2[31] + draft2[32] + draft2[33] + draft2[34] + draft2[35] + draft2[36] + draft2[37] + draft2[38] + draft2[39] + draft2[40] + draft2[41] + draft2[42] + draft2[43] + draft2[44] + draft2[45] + draft2[46] + draft2[47] + draft2[48] + draft2[49] + draft2[50] + draft2[51] + draft2[52] + draft2[53] + draft2[54] + draft2[55] + draft2[56] + draft2[57] + draft2[58] + draft2[59] + draft2[60] + draft2[61] + draft2[62] + draft2[63] + draft2[64] + draft2[65] + draft2[66] + draft2[67] + draft2[68] + draft2[69] + draft2[70] + draft2[71] + draft2[72] + draft2[73] + draft2[74] + draft2[75] + draft2[76] + draft2[77] + draft2[78] + draft2[79] + draft2[80] + draft2[81] + draft2[82] + draft2[83] + draft2[84] + draft2[85] + draft2[86] + draft2[87] + draft2[88] + draft2[89] 

Total number of motorcycle page to be scrapped.

In [207]:
len(motorcycle)

1635

I close and open a new chrome window so I won't get any error for 'TimeoutException'.

In [212]:
driver.close()
driver = webdriver.Chrome(ChromeDriverManager().install())

/var/folders/bq/1n8bgy_57l74n18m4pnvpyj00000gn/T/ipykernel_97374/2765047397.py:2: DeprecationWarning: executable_path has been deprecated, please pass in a Service object
  driver = webdriver.Chrome(ChromeDriverManager().install())


Putting it as mc_page so I won't get lost along the way.

In [209]:
mc_page = motorcycle

In [210]:
len(mc_page)

1635

The for loop that is used to scrape the data., creating a list(mc_list) so the scrap items can be put in it.

In [235]:
mc_list = []

Extracting the 10 items now, will remove the along the way if its not needed.

In [290]:
count = 742
for link in mc_page[742:]:
    mc_dict = {}
    go_to_page(link)

    source = requests.get(link)
    page = BeautifulSoup(source.content, "html.parser") 
    
    motor = page.find_all(name='h2', class_='text-center')
    motor_name = motor[0].get_text().strip()
    motor_price = motor[1].get_text().strip()

    status = page.find_all(name='td',class_='value')        
    listing_type = status[0].get_text().strip()
    brand = status[1].get_text().strip()
    model = status[2].get_text().strip()
    engine_cap = status[3].get_text().strip()
    classification = status[4].get_text().strip()
    reg_date = status[5].get_text().strip()
    coe_exp_date = status[6].get_text().strip()
    mileage = status[7].get_text().strip()
    no_of_owners = status[8].get_text().strip()
    type_of_vehicle = status[9].get_text().strip()

    # Storing data into dictionary
    mc_dict['motor_name'] = motor_name
    mc_dict['motor_price'] = motor_price
    mc_dict['listing_type'] = listing_type
    mc_dict['brand'] = brand
    mc_dict['model'] = model
    mc_dict['engine_cap'] = engine_cap
    mc_dict['classification'] = classification
    mc_dict['reg_date'] = reg_date
    mc_dict['coe_exp_date'] = coe_exp_date
    mc_dict['mileage'] = mileage
    mc_dict['no_of_owners'] = no_of_owners
    mc_dict['type_of_vehicle'] = type_of_vehicle
    

    # append the dictionary into a list 
    mc_list.append(mc_dict)
    print(f"{count}/{(len(motorcycle)-1)}")
    count += 1

742/1634


IndexError: list index out of range

Creating a new dictionary for every page for easy convert to a dataframe.

It stopped at list number 742 and thats where I realised the page format for new bikes is different so I had to make another for loop for the new bikes links.

In [359]:
count = 0
for link in mc_page[:]:
    mc_dict = {}
    go_to_page(link)

    source = requests.get(link)
    page = BeautifulSoup(source.content, "html.parser") 
    
    motor = page.find(name='h2', class_='mb-3 mt-3')
    motor_name = motor.get_text().strip()

    motor_price = 0

    status = page.find_all(name='td',class_='value')        
    listing_type = '-'
    brand = status[0].get_text().strip()
    model = status[1].get_text().strip()
    engine_cap = status[4].get_text().strip()
    classification = status[3].get_text().strip()
    reg_date = '00/00/00'
    coe_exp_date = '00/00/00'
    mileage = '0km'
    no_of_owners = 0
    type_of_vehicle = status[2].get_text().strip()

    # Storing data into dictionary
    mc_dict['motor_name'] = motor_name
    mc_dict['motor_price'] = motor_price
    mc_dict['listing_type'] = listing_type
    mc_dict['brand'] = brand
    mc_dict['model'] = model
    mc_dict['engine_cap'] = engine_cap
    mc_dict['classification'] = classification
    mc_dict['reg_date'] = reg_date
    mc_dict['coe_exp_date'] = coe_exp_date
    mc_dict['mileage'] = mileage
    mc_dict['no_of_owners'] = no_of_owners
    mc_dict['type_of_vehicle'] = type_of_vehicle
    

    # append the dictionary into a list 
    mc_list.append(mc_dict)
    print(f"{count}/{(len(motorcycle)-1)}")
    count += 1

1353/1634
1354/1634
1355/1634
1356/1634
1357/1634
1358/1634
1359/1634
1360/1634
1361/1634
1362/1634
1363/1634
1364/1634
1365/1634
1366/1634
1367/1634
1368/1634
1369/1634
1370/1634
1371/1634
1372/1634
1373/1634
1374/1634
1375/1634
1376/1634
1377/1634
1378/1634
1379/1634
1380/1634
1381/1634
1382/1634
1383/1634
1384/1634
1385/1634
1386/1634
1387/1634
1388/1634
1389/1634
1390/1634
1391/1634
1392/1634
1393/1634
1394/1634
1395/1634
1396/1634
1397/1634
1398/1634
1399/1634
1400/1634
1401/1634
1402/1634
1403/1634
1404/1634
1405/1634
1406/1634
1407/1634
1408/1634
1409/1634
1410/1634
1411/1634
1412/1634
1413/1634
1414/1634
1415/1634
1416/1634
1417/1634
1418/1634
1419/1634
1420/1634
1421/1634
1422/1634
1423/1634
1424/1634
1425/1634
1426/1634
1427/1634
1428/1634
1429/1634
1430/1634
1431/1634
1432/1634
1433/1634
1434/1634
1435/1634
1436/1634
1437/1634
1438/1634
1439/1634
1440/1634
1441/1634
1442/1634
1443/1634
1444/1634
1445/1634
1446/1634
1447/1634
1448/1634
1449/1634
1450/1634
1451/1634
1452/1634


For new bikes they do not have afew items listed in used bikes page so I had to assign them as thei minimum amount as default.

In [360]:
len(mc_list)

1635

Closing the chrome when I have all the items collected.

In [361]:
driver.close()

Creating the dataframe.

In [ ]:
mc_df = pd.DataFrame(mc_list)

In [42]:
mc_df

motor_name motor_price listing_type   brand  \
0                   BMW R1250RT   SGD$68000      Paid Ad     BMW   
1     Honda VFR800X CrossRunner   SGD$16800      Paid Ad   Honda   
2                 Yamaha XSR900   SGD$28888      Paid Ad  Yamaha   
3                  Honda PCX150    SGD$9222      Paid Ad   Honda   
4              Honda Shadow 400   SGD$12800      Paid Ad   Honda   
...                         ...         ...          ...     ...   
1630                Zontes T310           0            -  Zontes   
1631                Zontes V310           0            -  Zontes   
1632                Zontes X310           0            -  Zontes   
1633              Zontes ZT150G           0            -  Zontes   
1634              Zontes ZT150U           0            -  Zontes   

                          model engine_cap classification    reg_date  \
0                   BMW R1250RT     1254cc        Class 2  27/10/2021   
1     Honda VFR800X CrossRunner      782cc        Class 2  23/05/2015   
2                 Yamaha XSR900      847cc        Class 2  02/12/2020   
3                  Honda PCX150      153cc       Class 2B  29/08/2017   
4              Honda Shadow 400      398cc       Class 2A  23/01/1998   
...                         ...        ...            ...         ...   
1630                Zontes T310      312cc       Class 2A    00/00/00   
1631                Zontes V310      312cc       Class 2A    00/00/00   
1632                Zontes X310      312cc       Class 2A    00/00/00   
1633              Zontes ZT150G      150cc       Class 2B    00/00/00   
1634              Zontes ZT150U      150cc       Class 2B    00/00/00   

                                  coe_exp_date  mileage no_of_owners  \
0      26/10/2031 (8yrs 9mths 25days COE left)   2731km            1   
1      22/05/2025 (2yrs 4mths 21days COE left)  51000km            2   
2            01/12/2030 (7yrs 11mths COE left)    841km            1   
3      28/08/2027 (4yrs 7mths 27days COE left)        -            -   
4     30/11/2027 (4yrs 10mths 29days COE left)   9999km            -   
...                                        ...      ...          ...   
1630                                  00/00/00      0km            0   
1631                                  00/00/00      0km            0   
1632                                  00/00/00      0km            0   
1633                                  00/00/00      0km            0   
1634                                  00/00/00      0km            0   

     type_of_vehicle  
0      Sport Tourers  
1      Sport Tourers  
2       Street Bikes  
3           Scooters  
4            Cruiser  
...              ...  
1630   Sport Tourers  
1631    Street Bikes  
1632   Sport Tourers  
1633    Street Bikes  
1634    Street Bikes  

[1635 rows x 12 columns]

In [43]:
mc_df.to_csv('Dataset_Nurul Liyana.csv')

Saving the dataframe

## BIKEMART.SG

Since I had no prices in the new bikes pages, I went to search for prices in Bikemart.sg

In [544]:
source = requests.get('https://bikemart.sg/motorcycle-price-list/')
page = BeautifulSoup(source.content, "html.parser") 

For this website, I dont have to go page by page to extract, I could get all the items in one find, which was easier too. I only had to change the name of motorcycle names as it was it wasn't the same format as the website above. 

In [673]:
motor_list = []
price_list = []
year_list = []

In [674]:
motor = page.find_all(name='td')
for item in motor:
    motorprice = item.get_text().strip()

    if motorprice != '':
        if ' (Used)' in motorprice:
            motor_name = motorprice.replace(' (Used)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' New (50th Anniversary Edition)' in motorprice:
            motor_name = motorprice.replace(' New (50th Anniversary Edition)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' New (Special Edition)' in motorprice:
            motor_name = motorprice.replace(' New (Special Edition)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' New (Bikers)' in motorprice:
            motor_name = motorprice.replace(' New (Bikers)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' New (Ohlins)' in motorprice:
            motor_name = motorprice.replace(' New (Ohlins)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' (New) (Bikers & Ohlins)' in motorprice:
            motor_name = motorprice.replace(' (New) (Bikers & Ohlins)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' (New) (Bikers)' in motorprice:
            motor_name = motorprice.replace(' (New) (Bikers)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' (Ohlins)' in motorprice: 
            motor_name = motorprice.replace(' (New) (Ohlins)','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' New' in motorprice:
            motor_name = motorprice.replace(' New','')
            year = motor_name[-4:]
            year_list.append(year)
            name = motor_name.replace(year,'')
            motor_list.append(name)
        elif ' (New)' in motorprice:
            motor_name = motorprice.replace(' (New)','')
            Abs = motor_name[-4:]
            num = motor_name[-2:]
            if Abs == ' ABS':
                year = 2022
                year_list.append(year)
                name = motor_name
            elif Abs == '-ABS':
                year = 1978
                year_list.append(year)
                name = motor_name
            elif num == '14':
                year = 2014
                year_list.append(year)
                name = motor_name
            else:
                year = motor_name[-4:]
                year_list.append(year)
                name = motor_name.replace(year,'')
            motor_list.append(name)
        elif '$' in motorprice:
            price = motorprice.replace('$','').replace(',','').replace('.00','')
            if len(price) >= 8:
                space = price[-5:]
                if space[0] == '1':
                    back = price[-5:] 
                    final = price.replace(price,back)
                    price_list.append(final)
                else:
                    back = price[-4:] 
                    final = price.replace(price,back)
                    price_list.append(final)
            else:
                price_list.append(price)

In [677]:
print(len(motor_list))
print(len(price_list))
print(len(year_list))

633
633
633


The total length of motorcycle and its prices is 633.

In [678]:
motor_price_df = pd.DataFrame()
motor_price_df['motor_name'] = motor_list
motor_price_df['motor_price'] = price_list
motor_price_df['year'] = year_list

In [679]:
motor_price_df

motor_name motor_price  year
0                     Adiva AD200 ABS        14300  2016
1                     Adiva AD300 ABS        14500  2016
2                      Adiva AR 3 ABS        16100  2016
3       Aprilia Caponord 1200 ABS ADD        31000  2016
4     Aprilia Caponord 1200 Rally ABS        29000  2016
..                                 ...         ...   ...
628  Yamaha YZF-R155 (R15) ABS Manual        12500  2020
629          Yamaha YZF-R3 ABS Manual         9000  2019
630                Yamaha YZF-R6S ABS         8000  2008
631        Yamaha YZF155 (R15 V3) ABS         6500  2020
632        Yamaha YZF155 (R15 V4) ABS         7100  2022

[633 rows x 3 columns]

In [680]:
motor_price_df.to_csv('Supporting_Data_Nurul Liyana(5).csv')

Saving the dataframe

### There's one more scrapping website to be done, next Jupyter Notebook will be Project_Script_SD(2)